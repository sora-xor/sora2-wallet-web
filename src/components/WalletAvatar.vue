<template>
  <Identicon
    :size="size"
    :theme="theme"
    :value="address"
  />
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'vue-property-decorator'
import Identicon from '@polkadot/vue-identicon'

enum THEMES {
  POLKADOT = 'polkadot',
  SUBSTRATE = 'substrate',
  BEACHBALL = 'beachball'
}

@Component({
  components: {
    Identicon
  }
})
export default class WalletAvatar extends Vue {
  @Prop({ default: 32, type: Number }) readonly size!: number
  @Prop({ default: THEMES.POLKADOT, type: String }) readonly theme!: string
  @Prop({ default: '', type: String, required: true }) readonly address!: string
}
</script>
