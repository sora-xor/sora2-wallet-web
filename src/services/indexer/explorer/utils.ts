import { formatStringNumber } from '../../../util';

import type { SubqueryAssetEntity } from '../subquery/types';
import type { SubsquidAssetEntity } from '../subsquid/types';
import type { FiatPriceObject, UpdatesStream, PoolApyObject } from '../types';

export function parseAssetFiatPrice(entity: SubsquidAssetEntity | SubqueryAssetEntity): FiatPriceObject {
  const acc = {};
  const id = entity.id;
  const priceFPNumber = formatStringNumber(entity.priceUSD);
  const isPriceFinity = priceFPNumber.isFinity();
  if (isPriceFinity) {
    acc[id] = priceFPNumber.toCodecString();
  }
  return acc;
}

export function parsePriceStreamUpdate(entity: UpdatesStream): Nullable<FiatPriceObject> {
  if (!entity?.data) return null;

  const data = JSON.parse(entity.data);

  return Object.entries(data).reduce((acc, [id, price]) => {
    const priceFPNumber = formatStringNumber(price as string);
    const isPriceFinity = priceFPNumber.isFinity();
    if (isPriceFinity) {
      acc[id] = priceFPNumber.toCodecString();
    }
    return acc;
  }, {});
}

export function parseApyStreamUpdate(entity: UpdatesStream): PoolApyObject {
  const data = entity?.data ? JSON.parse(entity.data) : {};

  return Object.entries(data).reduce((acc, [id, apy]) => {
    const strategicBonusApyFPNumber = formatStringNumber(apy as string);
    const isStrategicBonusApyFinity = strategicBonusApyFPNumber.isFinity();
    if (isStrategicBonusApyFinity) {
      acc[id] = strategicBonusApyFPNumber.toCodecString();
    }
    return acc;
  }, {});
}
