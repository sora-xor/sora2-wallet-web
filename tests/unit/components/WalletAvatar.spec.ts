import { shallowMount } from '@vue/test-utils'
import Identicon from '@polkadot/vue-identicon'

import { useDescribe, localVue } from '../../utils'
import { MOCK_WALLET_AVATAR } from '../../utils/WalletAvatarMock'

import WalletAvatar from '@/components/WalletAvatar.vue'

useDescribe('WalletAvatar.vue', WalletAvatar, () => {
  MOCK_WALLET_AVATAR.map(item => it(`[${item.title}]: should be rendered correctly`, () => {
    const propsData = {
      size: item.size,
      theme: item.theme,
      address: item.address
    }
    const wrapper = shallowMount(WalletAvatar, {
      localVue,
      propsData,
      stubs: {
        Identicon: Identicon
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  }))
})
