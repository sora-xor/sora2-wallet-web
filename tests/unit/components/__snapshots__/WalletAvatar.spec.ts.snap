// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletAvatar.vue [With Beachball Theme]: should be rendered correctly 1`] = `<!--(a, b, c, d) =&gt; createElement$1(vm, a, b, c, d, true)-->`;

exports[`WalletAvatar.vue [With Empty Address]: should be rendered correctly 1`] = `
<svg
  height="32"
  key=""
  viewBox="0 0 64 64"
  width="32"
>
  <circle
    cx="32"
    cy="32"
    fill="#eee"
    r="32"
  />
  <circle
    cx="32"
    cy="8"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="32"
    cy="20"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="21.607695154586736"
    cy="14"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="11.215390309173472"
    cy="20"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="21.607695154586736"
    cy="26"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="11.215390309173472"
    cy="32"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="11.215390309173472"
    cy="44"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="21.607695154586736"
    cy="38"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="21.607695154586736"
    cy="50"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="32"
    cy="56"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="32"
    cy="44"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="42.392304845413264"
    cy="50"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="52.78460969082653"
    cy="44"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="42.392304845413264"
    cy="38"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="52.78460969082653"
    cy="32"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="52.78460969082653"
    cy="20"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="42.392304845413264"
    cy="26"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="42.392304845413264"
    cy="14"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="32"
    cy="32"
    fill="#ddd"
    r="5"
  />
</svg>
`;

exports[`WalletAvatar.vue [With Size 10]: should be rendered correctly 1`] = `
<svg
  height="10"
  key=""
  viewBox="0 0 64 64"
  width="10"
>
  <circle
    cx="32"
    cy="32"
    fill="#eee"
    r="32"
  />
  <circle
    cx="32"
    cy="8"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="32"
    cy="20"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="21.607695154586736"
    cy="14"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="11.215390309173472"
    cy="20"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="21.607695154586736"
    cy="26"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="11.215390309173472"
    cy="32"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="11.215390309173472"
    cy="44"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="21.607695154586736"
    cy="38"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="21.607695154586736"
    cy="50"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="32"
    cy="56"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="32"
    cy="44"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="42.392304845413264"
    cy="50"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="52.78460969082653"
    cy="44"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="42.392304845413264"
    cy="38"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="52.78460969082653"
    cy="32"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="52.78460969082653"
    cy="20"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="42.392304845413264"
    cy="26"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="42.392304845413264"
    cy="14"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="32"
    cy="32"
    fill="#ddd"
    r="5"
  />
</svg>
`;

exports[`WalletAvatar.vue [With Substrate Theme]: should be rendered correctly 1`] = `
<substrate
  address=""
  key=""
  size="32"
/>
`;

exports[`WalletAvatar.vue [With default Theme (Polkadot) and Size]: should be rendered correctly 1`] = `
<svg
  height="32"
  key=""
  viewBox="0 0 64 64"
  width="32"
>
  <circle
    cx="32"
    cy="32"
    fill="#eee"
    r="32"
  />
  <circle
    cx="32"
    cy="8"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="32"
    cy="20"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="21.607695154586736"
    cy="14"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="11.215390309173472"
    cy="20"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="21.607695154586736"
    cy="26"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="11.215390309173472"
    cy="32"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="11.215390309173472"
    cy="44"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="21.607695154586736"
    cy="38"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="21.607695154586736"
    cy="50"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="32"
    cy="56"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="32"
    cy="44"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="42.392304845413264"
    cy="50"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="52.78460969082653"
    cy="44"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="42.392304845413264"
    cy="38"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="52.78460969082653"
    cy="32"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="52.78460969082653"
    cy="20"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="42.392304845413264"
    cy="26"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="42.392304845413264"
    cy="14"
    fill="#ddd"
    r="5"
  />
  <circle
    cx="32"
    cy="32"
    fill="#ddd"
    r="5"
  />
</svg>
`;
